src/sqlitecalendarstorage.cpp

```diff
@@ -0,0 +1,255 @@
+    Notebook::Ptr mDbNotebook;
+    bool mIsDefault = false;
+
+    bool loadNotebook(const QString &notebookUid);
+};
+//@endcond
+
+SqliteCalendarStorage::SqliteCalendarStorage(const MemoryCalendar::Ptr &cal,
+                                             const QString &databaseName)
+    : CalendarStorage(cal)
+    , d(new Private(databaseName))
+{
+}
+
+SqliteCalendarStorage::SqliteCalendarStorage(const QString &uid,
+                                             const QString &databaseName)
+    : CalendarStorage(uid)
+    , d(new Private(databaseName))
+{
+}
+
+SqliteCalendarStorage::~SqliteCalendarStorage()
+{
+    delete d;
+}
+
+bool SqliteCalendarStorage::open()
+{
+    if (!d->mBackend.open() || !d->loadNotebook(calendar()->id())) {
+        return false;
+    }
+    connect(&d->mBackend, &SingleSqliteBackend::modified,
+            this, &SqliteCalendarStorage::onModified);
+    connect(&d->mBackend, &SingleSqliteBackend::updated,
+            this, &SqliteCalendarStorage::onUpdated);
+    return CalendarStorage::open();
+}
+
+bool SqliteCalendarStorage::close()
+{
+    if (!d->mBackend.close()) {
+        return false;
+    }
+    d->mDbNotebook.clear();
+    d->mIsDefault = false;
+    disconnect(&d->mBackend, &SingleSqliteBackend::modified,
+               this, &SqliteCalendarStorage::onModified);
+    disconnect(&d->mBackend, &SingleSqliteBackend::updated,
+               this, &SqliteCalendarStorage::onUpdated);
+    return CalendarStorage::close();
+}
+
+bool SqliteCalendarStorage::Private::loadNotebook(const QString &notebookUid)
+{
+    mDbNotebook.clear();
+    mIsDefault = false;
+
+    Notebook::List list;
+    Notebook::Ptr defaultNb;
+    if (mBackend.notebooks(&list, &defaultNb)) {
+        for (Notebook::List::ConstIterator it = list.constBegin();
+             it != list.constEnd(); it++) {
+            if ((*it)->uid() == notebookUid) {
+                mDbNotebook = *it;
+                break;
+            }
+        }
+        mIsDefault = mDbNotebook && defaultNb
+            && mDbNotebook->uid() == defaultNb->uid();
+        return true;
+    } else {
+        qCWarning(lcMkcal) << "cannot load notebooks.";
+        return false;
+    }
+}
+
+void SqliteCalendarStorage::onModified()
+{
+    d->loadNotebook(calendar()->id());
+    emitStorageModified();
+}
+
+void SqliteCalendarStorage::onUpdated(const QHash<QString, QStringList> &added,
+                                      const QHash<QString, QStringList> &modified,
+                                      const QHash<QString, QStringList> &deleted)
+{
+    emitStorageUpdated(added.value(calendar()->id()),
+                       modified.value(calendar()->id()),
+                       deleted.value(calendar()->id()));
+}
+
+const Notebook::Ptr SqliteCalendarStorage::loadedNotebook() const
+{
+    return d->mDbNotebook;
+}
+
+bool SqliteCalendarStorage::erase()
+{
+    if (notebook() && d->mBackend.deleteNotebook(*notebook())) {
```

Comment:
I'm not sure about deleting notebooks by "erasing" storage. It feels kind of misleading when the real storage (e.g. a database file)  is not deleted, and commonly there's no use-case to actually need such removal.

Reply:
As discussed on IRC, I've removed completely the `erase()` routine from the CalendarStorage API.

Since it is used by sync plugins to delete notebooks because of upstream deletion for instance, I've instead created a static Notebook method that delete a "system" notebook.